src/app/core/store/documents/documents.effects.ts

```diff
@@ -155,4 +181,59 @@ export class DocumentsEffects {
               private store$: Store<AppState>) {
   }
 
+  private createAddCollectionAttributesAction(collection: CollectionModel, document: DocumentModel): Action {
+    const collectionCopy = {...collection, documentsCount: collection.documentsCount + 1};
+
+    const addedAttributes = Object.keys(document.data);
+
+    collectionCopy.attributes = this.updateAttributesInColection(collectionCopy.attributes, addedAttributes, []);
+
+    return new CollectionsAction.UpdateSuccess({collection: collectionCopy});
+  }
+
+  private createUpdateCollectionAttributesAction(collection: CollectionModel, document: DocumentModel, oldDocument: DocumentModel): Action {
+    const collectionCopy = {...collection};
+
+    const oldAttributesIds = Object.keys(oldDocument.data);
+    const newAttributesIds = Object.keys(document.data);
+
+    const addedAttributesIds = newAttributesIds.filter(attr => !oldAttributesIds.includes(attr));
+    const removedAttributesIds = oldAttributesIds.filter(attr => !newAttributesIds.includes(attr));
+
+    collectionCopy.attributes = this.updateAttributesInColection(collectionCopy.attributes, addedAttributesIds, removedAttributesIds);
+
+    return new CollectionsAction.UpdateSuccess({collection: collectionCopy});
+  }
+
+  private createRemoveCollectionsAttributesAction(collection: CollectionModel, document: DocumentModel): Action {
+    const collectionCopy = {...collection, documentsCount: Math.max(collection.documentsCount - 1, 0)};
+
+    const deletedAttributes = Object.keys(document.data);
+
+    collectionCopy.attributes = this.updateAttributesInColection(collectionCopy.attributes, [], deletedAttributes);
+
+    return new CollectionsAction.UpdateSuccess({collection: collectionCopy});
+  }
+
+  private updateAttributesInColection(attributes: AttributeModel[], attributeToInc: string[], attributeToDec: string[]): AttributeModel[] {
+    const newAttributes = attributes.map(attribute => {
+        if (attributeToDec.includes(attribute.id)) {
+          return {...attribute, usageCount: Math.max(attribute.usageCount - 1, 0)};
+        } else {
+          const attributeIndex = attributeToInc.indexOf(attribute.id);
+          if (attributeIndex !== -1) {
+            attributeToInc.splice(attributeIndex, 1);
+            return {...attribute, usageCount: attribute.usageCount + 1};
+          }
+        }
+        return attribute;
+      }
+    );
+
+    attributeToInc.map(attributeId => ({id: attributeId, name: attributeId, constraints: [], usageCount: 1}))
```

Comment:
Be aware that `name` does not always equal `id`. I would recommend using `extractAttributeName` from attribute utils here. After the attribute model refactoring there will no longer be this problem. But for now, the current behavior may break the table.